src/metalookup/features/extract_from_files.py

```diff
@@ -203,7 +203,30 @@ async def task(url: str) -> Optional[str]:
         return {file for file in extractable_files if file is not None}
 
     @staticmethod
-    def _get_extractable_files(links: list[str]) -> set[str]:
-        file_extensions = [os.path.splitext(link)[-1] for link in links]
+    async def _get_extractable_files(content: Content) -> set[str]:
+        """
+        Search through the html and extract all links that end in docx or pdf.
+        Filters out email, phone, javascript, or section links (see https://www.w3schools.com/tags/att_a_href.asp)
 
-        return {file for file, extension in zip(links, file_extensions) if extension in [".docx", ".pdf"]}
+        Note: Extracted links will all be absolute, i.e. relative links will be modified to be valid URLs.
+        """
+        soup = await content.soup()
+
+        def filter(link: str) -> bool:
+            correct_extension = any(link.endswith(ext) for ext in (".docx", ".pdf"))
+            wrong_kind = any(link.startswith(s) for s in ("#", "javascript", "tel", "mailto"))
+            return correct_extension and not wrong_kind
+
+        def process(link: str) -> str:
+            if link.startswith("http"):
+                return link  # we are good.
+            else:
+                url = urlparse(content.url)
+                link = link.strip("/")
+                if url.path != "":  # avoid multiple // in generated link...
+                    path = url.path.strip("/")
+                    return f"{url.scheme}://{url.netloc}/{path}/{link}"
+                else:
+                    return f"{url.scheme}://{url.netloc}/{link}"
+
+        return {process(link.get("href")) for link in soup.find_all(name="a", href=filter)}
```

Comment:
Interesting solution. href=filter calls the filter function?

Reply:
yep, the filters can be passed as kwargs where the kwarg key corresponds to the html tag attribute.